### PR TITLE
docs: Add handcrafted documentation

### PR DESCRIPTION
**Pull request by tmac1973**
This documentation augments the autogenerated API docs.

### CODE REVIEW

File: docs-source/static/authentication.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
+==================================
+Lacework Python SDK Authentication
+==================================
+
+Once you have installed the SDK you will need to determine how you
+want to pass authentication information to the SDK. There are a
+variety of ways to do this however they all require a valid API key
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:7]:
```suggestion
variety of ways to do this, however they all require a valid API key
```

File: docs-source/static/authentication.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+for the Lacework account you wish to access with the SDK. Note that
+API keys are generated at the account level only, not a the organization level.
+You must have/generate a key for each account you need access to.
+
+To read more about generating Lacework API keys read the Lacework docs
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:12]:
```suggestion
Learn more about generating Lacework API keys
```

File: docs-source/static/authentication.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+`here <https://docs.lacework.net/api/api-access-keys-and-tokens>`_.
+
+If you aren't sure what a Lacework organization is you can read `here <https://docs.lacework.net/console/organization-overview>`_.
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:15]:
```suggestion
Learn more about Lacework organizations `here <https://docs.lacework.net/console/organization-overview>`_.
```

File: docs-source/static/authentication.rst [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+Once you have created an API key in the Lacework console you should download
+the JSON file containing your new API credentials. This will contain 3 or 4 properties
+depending on whether the keys you generated are part of a Lacework organization. These
+properties are will be ``keyId``, ``secret``, ``account``, and optionally ``subaccount``
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:20]:
```suggestion
properties are will be ``keyId``, ``secret``, ``account``, and optionally ``subaccount``.
```

File: docs-source/static/authentication.rst [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+(if your key comes from a Lacework organization the the ``account`` property represents the
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:21]:
```suggestion
 If your key comes from a Lacework organization the the ``account`` property represents the
```

File: docs-source/static/authentication.rst [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+name of the organization and the ``subaccount`` property represents that name of the account
+within that organization.)
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:23]:
```suggestion
within that organization.
```

File: docs-source/static/authentication.rst [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+Be sure keep these credentials **SECURE**!
+
+Now that you have this file you can choose which method of authentication to use:
+
+Auth Method 1: Install the Lacework CLI
+=======================================
+
+The Lacework Python SDK does **NOT** require that you install the Lacework CLI tool. However if you do
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:32]:
```suggestion
The Lacework Python SDK does **NOT** require that you install the Lacework CLI tool. However, if you do
```

File: docs-source/static/authentication.rst [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+have the Lacework CLI `installed <https://docs.lacework.net/cli#install-the-lacework-cli>`_ AND
+`configured <https://docs.lacework.net/cli#configure-the-cli>`_ then as part of it's configuration
+process it will create a file in your home directory called ``.lacework.toml`` which will contain
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:35]:
```suggestion
process the CLI will create a file in your home directory called ``.lacework.toml``. This file has
```

File: docs-source/static/authentication.rst [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+one or more sections containing Lacework credentials, with each section representing creds for a
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:36]:
```suggestion
one or more sections containing Lacework credentials, with each section representing information for a
```

File: docs-source/static/authentication.rst [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+single Lacework account.
+
+If you have this file in place and you have not specified credentials using any other method then
+the Python SDK willuse the credentials stored there to access the Lacework API endpoints. Note that by
```
Comment (ipcrm) [inline docs-source/static/authentication.rst:40]:
```suggestion
the Python SDK will use those credentials. Note, by
```

File: docs-source/static/timestamps.rst [hunk 11]
```diff
@@ -0,0 +1,40 @@
+========================================
+Lacework Python SDK Timestamp Generation
+========================================
+
+For all "search" methods Lacework requires ``start_time`` and ``end_time`` arguments which
+are used to specify the search window. Additionally some "get" methods also require them.
```
Comment (ipcrm) [inline docs-source/static/timestamps.rst:6]:
```suggestion
are used to specify the search window. Additionally, some "get" methods also require them.
```

File: docs-source/static/timestamps.rst [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+These must be specified as strings in the following format:
+
+    ``"%Y-%m-%dT%H:%M:%S%z"``
+
+For example:
```
Comment (ipcrm) [inline docs-source/static/timestamps.rst:11]:
```suggestion
Example:
```

File: docs-source/static/timestamps.rst [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+    ``"2024-01-08T22:34:10+0000"``
+
+You are free to generate these strings however you like but you may find it useful to
```
Comment (ipcrm) [inline docs-source/static/timestamps.rst:15]:
```suggestion
You are free to generate these strings however you like, but you may find it useful to
```

File: docs-source/static/timestamps.rst [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+use the following function (or something similar.)
```
Comment (ipcrm) [inline docs-source/static/timestamps.rst:16]:
```suggestion
use the following function (or something similar).
```

File: docs-source/static/usage.rst [hunk 15]
```diff
@@ -0,0 +1,123 @@
+=========================
+Lacework Python SDK Usage
+=========================
+
+The Lacework Python SDK closely mirrors the `Lacework API <https://docs.lacework.net/api/v2/docs/>`_
+in structure. The main class of the SDK is ``LaceworkClient`` which has attributes representing
```
Comment (ipcrm) [inline docs-source/static/usage.rst:6]:
```suggestion
in structure. The main class of the SDK is ``LaceworkClient`` which has classes representing
```

Right?

File: docs-source/static/usage.rst [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+the various resources/endpoints of the Lacework API. Depending on the resource these attributes
```
Comment (ipcrm) [inline docs-source/static/usage.rst:7]:
```suggestion
the various resources/endpoints of the Lacework API. Depending on the resource, these classes
```

File: docs-source/static/usage.rst [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+will have some combination of CRUD and search methods, though a few have other unique methods.
+You can read about the various attributes and methods of the SDK `here <https://lacework.github.io/python-sdk/autoapi/laceworksdk/index.html>`_.
+
+Usage Examples
+==============
+
+Example 1: Create a New User
+----------------------------
+
+This example leverages the ``team_users`` `attribute <https://lacework.github.io/python-sdk/autoapi/laceworksdk/api/v2/team_users/index.html>`_
```
Comment (ipcrm) [inline docs-source/static/usage.rst:17]:
```suggestion
This example leverages the ``team_users`` `class <https://lacework.github.io/python-sdk/autoapi/laceworksdk/api/v2/team_users/index.html>`_
```

File: docs-source/static/usage.rst [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+
+    from laceworksdk import LaceworkClient
+
+    lw = LaceworkClient(profile="default")
+    response = lw.team_users.create("testuser", "testuser@testdomain.com", "Test Company")
+
+Example 2: Searching for an Alert
+---------------------------------
+
+This example leverages the ``alerts`` `attribute <https://lacework.github.io/python-sdk/autoapi/laceworksdk/api/v2/alerts/index.html>`_
```
Comment (ipcrm) [inline docs-source/static/usage.rst:32]:
```suggestion
This example leverages the ``alerts`` `class <https://lacework.github.io/python-sdk/autoapi/laceworksdk/api/v2/alerts/index.html>`_
```

File: docs-source/static/usage.rst [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+to find all CRITICAL alerts that occured in the last 24 hours.
```
Comment (ipcrm) [inline docs-source/static/usage.rst:33]:
```suggestion
to find all ``CRITICAL`` alerts that occured in the last 24 hours.
```

File: docs-source/static/usage.rst [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+
+Note: This search will return a generator object, not a list or dict. If you are
+unfamiliar with Python generators you may want to read up on them.
+
+https://wiki.python.org/moin/Generators
+
+Example 3: Retrieving a Compliance Report
+-----------------------------------------
+
+This example leverages the ``cloud_accounts`` `attribute <https://lacework.github.io/python-sdk/autoapi/laceworksdk/api/v2/cloud_accounts/index.html>`_
```
Comment (ipcrm) [inline docs-source/static/usage.rst:82]:
```suggestion
This example leverages the ``cloud_accounts`` `class <https://lacework.github.io/python-sdk/autoapi/laceworksdk/api/v2/cloud_accounts/index.html>`_
```

File: docs-source/static/usage.rst [hunk 21]
```diff
@@ -0,0 +1,123 @@
...
+first to retrieve a list of AWS account integrations, specifically those that retrieve "config"
+information. It uses a python list comprehension to select the first one it finds and parses the
+AWS account ID from that data.
+
+Once it has a valid AWS account integration it uses this to pull a CIS 1.4 report using the
+``reports`` `attribute <https://lacework.github.io/python-sdk/autoapi/laceworksdk/api/v2/reports/index.html>`_.
```
Comment (ipcrm) [inline docs-source/static/usage.rst:88]:
```suggestion
``reports`` `class <https://lacework.github.io/python-sdk/autoapi/laceworksdk/api/v2/reports/index.html>`_.
```

**Review by ipcrm — CHANGES_REQUESTED** [review]
Amazing! I have a few thoughts on language changes, but they are all suggestions!

File: docs-source/static/usage.rst (hunk 21 above)
Comment (tmac1973) [reply docs-source/static/usage.rst:88]: Heh I went back and forth on this.

**Review by ipcrm — APPROVED** [review]

**Pull request merged**